### ISSUE TITLE
RP2040 audiobusio,PDMIN not working with latest build

### ISSUE THREAD

**Issue by jerryneedell** [adafruit/circuitpython#5967]
### CircuitPython version

```python
built from tip of main

Adafruit CircuitPython 7.2.0-alpha.1-274-g5435a40c5 on 2022-02-02; Adafruit QT Py RP2040 with rp2040
```

### Code/REPL

```python
Ran demo code from https://learn.adafruit.com/neopixel-mini-vu-meter/coding-the-neopixel-mini-vu-meter

uncommneted line to print PDM Mic readings.


# SPDX-FileCopyrightText: 2022 Liz Clark for Adafruit Industries
# SPDX-License-Identifier: MIT

import time
import array
import math
import board
import audiobusio
import simpleio
import neopixel

#  neopixel setup
pixel_pin = board.A0

pixel_num = 16

pixels = neopixel.NeoPixel(pixel_pin, pixel_num, brightness=0.1, auto_write=False)

#  function to average mic levels
def mean(values):
    return sum(values) / len(values)

#  function to return mic level
def normalized_rms(values):
    minbuf = int(mean(values))
    samples_sum = sum(
        float(sample - minbuf) * (sample - minbuf)
        for sample in values
    )

    return math.sqrt(samples_sum / len(values))

#  mic setup
mic = audiobusio.PDMIn(board.TX,
                       board.A1, sample_rate=16000, bit_depth=16)
samples = array.array('H', [0] * 160)

#  variable to hold previous mic level
last_input = 0

#  neopixel colors
GREEN = (0, 127, 0)
RED = (127, 0, 0)
YELLOW = (127, 127, 0)
OFF = (0, 0, 0)

#  array of colors for VU meter
colors = [GREEN, GREEN, GREEN, GREEN, GREEN, GREEN, GREEN, GREEN,
          GREEN, YELLOW, YELLOW, YELLOW, YELLOW, YELLOW, RED, RED]

#  begin with pixels off
pixels.fill(OFF)
pixels.show()

while True:
    #  take in audio
    mic.record(samples, len(samples))
    #  magnitude holds the value of the mic level
    magnitude = normalized_rms(samples)
    #  uncomment to see the levels in the REPL
    print((magnitude,))

    #  mapping the volume range (125-500) to the 16 neopixels
    #  volume range is optimized for music. you may want to adjust the range
    #  based on the type of audio that you're trying to show
    mapped_value = simpleio.map_range(magnitude, 125, 500, 0, 16)
    #  converting the mapped range to an integer
    input_val = int(mapped_value)

    #  if the mic input has changed from the last input value...
    if last_input != input_val:
        for i in range(input_val):
            #  if the level is lower...
            if last_input > input_val:
                for z in range(last_input):
                    #  turn those pixels off
                    pixels[z] = (OFF)
            #  turn on pixels using the colors array
            pixels[i] = (colors[i])
            pixels.show()
            #  update last_input
            last_input = input_val

    time.sleep(0.01)
```
```


### Behavior

PDM mic returns all 0.0

### Description

_No response_

### Additional information

Works normally on 7.1.1

**Comment by DavePutz** [comment]
Looks like commit 3e13d4f (sideset_enable support) may be the culprit. I will continue to investigate.

**Comment by tannewt** [comment]
Thanks for looking into this @DavePutz !

**Comment by dhalbert** [comment]
Should be fixed by #5984. @jerryneedell are you able to retest with the latest build?

**Comment by jerryneedell** [comment]
Yes. Works fine. Thank you!
